src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Field.cs

```diff
@@ -12,11 +10,65 @@ public override BoundNode VisitFieldAccess(BoundFieldAccess node)
 
-            // field access of a tuple actually accesses the underlying field.
-            FieldSymbol field = (node.FieldSymbol as TupleFieldSymbol)?.UnderlyingTypleFieldSymbol ??
-                                node.FieldSymbol;
+            return RewriteTupleFieldAccess(node, rewrittenReceiver);
+        }
+
+        /// <summary>
+        /// Converts access to a tuple instance into access into the underlying ValueTuple(s).
+        ///
+        /// For instance, tuple.Item8
+        /// produces fieldAccess(field=Item1, receiver=fieldAccess(field=Rest, receiver=ValueTuple for tuple))
+        /// </summary>
+        private BoundExpression RewriteTupleFieldAccess(BoundFieldAccess node, BoundExpression rewrittenReceiver)
+        {
+            var tupleField = node.FieldSymbol as TupleFieldSymbol;
+            if (tupleField == null)
+            {
+                return MakeFieldAccess(node.Syntax, rewrittenReceiver, node.FieldSymbol, node.ConstantValue, node.ResultKind, node.Type, node);
+            }
+
+            TupleTypeSymbol tupleType = (TupleTypeSymbol)tupleField.ContainingSymbol;
+            NamedTypeSymbol underlyingTupleType = tupleType.UnderlyingTupleType;
+
+            int tupleRemainder;
+            int tupleChainLength = TupleTypeSymbol.NumberOfValueTuples(tupleType.Arity, out tupleRemainder);
+
+            int fieldRemainder;
+            int fieldChainLength = TupleTypeSymbol.NumberOfValueTuples(tupleField.Index + 1, out fieldRemainder);
+
+            BoundExpression currentField = rewrittenReceiver;
+            NamedTypeSymbol currentLinkType = underlyingTupleType;
+
+            for (int fieldLinkIndex = 0; fieldLinkIndex < fieldChainLength; fieldLinkIndex++)
+            {
+                if (fieldLinkIndex != 0)
+                {
+                    currentLinkType = TupleTypeSymbol.GetNestedTupleType(currentLinkType, 1);
+                }
+
+                int containingLinkSize = 8;
+                int indexInLink = 8;
+                if (fieldLinkIndex == fieldChainLength - 1)
+                {
+                    if (tupleChainLength == fieldChainLength)
+                    {
+                        containingLinkSize = tupleRemainder;
+                    }
+
+                    indexInLink = fieldRemainder;
+                }
+
+                WellKnownMember wellKnownTypeMember = TupleTypeSymbol.GetTupleTypeMember(containingLinkSize, indexInLink);
+                var linkField = (FieldSymbol)Binder.GetWellKnownTypeMember(_compilation, wellKnownTypeMember, _diagnostics, syntax: node.Syntax);
+
+                FieldSymbol nestedFieldSymbol = linkField?.AsMember(currentLinkType);
+
+                TypeSymbol nestedFieldType = currentLinkType.TypeArguments[indexInLink - 1];
+                currentField = new BoundFieldAccess(node.Syntax, currentField, nestedFieldSymbol, node.ConstantValue, node.ResultKind, nestedFieldType);
```

Comment:
We can't actually get `null` at that point because all the field accessors have been checked when constructing the symbol (see `GetTupleUnderlyingTypeAndFields`).
I'll change the code to check for `null` and return `node`, which is always safe.